core/server/worker/src/main/java/alluxio/worker/grpc/BlockWorkerClientServiceHandler.java

```diff
@@ -149,7 +151,11 @@ public BlockWorkerClientServiceHandler(WorkerProcess workerProcess, FileSystemCo
   @Override
   public void cache(CacheRequest request, StreamObserver<CacheResponse> responseObserver) {
     RpcUtils.call(LOG, () -> {
-      mBlockWorker.cache(request);
+      try {
+        mBlockWorker.cache(request);
+      } catch (ExecutionException | InterruptedException e) {
+        throw AlluxioStatusException.fromThrowable(e);
```

Comment:
Let's do this inside BlockWorker implementation and avoid throwing from there in the first place.
Also, in case of ExecutionException, we actually want to throw its cause (`ExecutionException#getCause()`) instead.

Reply:
since we throw alluxioException|IOException, so no need to catch exceptions in the service handler.